tests/unit/frameworks/continuous_learning/multihead/multihead_test.py

```diff
@@ -28,18 +28,38 @@
 import unittest
 
 from torch.utils.data import DataLoader
+from torchvision import transforms
 
-from nupic.research.frameworks.continuous_learning.multihead.subdataset import (
-    get_datasets,
+from nupic.research.frameworks.continuous_learning.multihead.multihead import (
+    get_active_classes,
 )
+from nupic.research.frameworks.pytorch.dataset_utils.dataset_utils import split_dataset
+from nupic.research.frameworks.pytorch.test_utils.fake_data import (
+    FakeDataPredefinedTargets,
+)
+
+
+def setup(scenario):
+    fake_train_dataset = FakeDataPredefinedTargets(image_size=(1, 28, 28))
+    fake_test_dataset = FakeDataPredefinedTargets(image_size=(1, 28, 28))
+
+    if scenario in ("task", "domain"):
+        target_transform = transforms.Lambda(lambda y: y % 2)
+    else:
+        target_transform = None
 
+    train_datasets = split_dataset(fake_train_dataset,
+                                   groupby=lambda x: x[1] // 2,
+                                   target_transform=target_transform)
```

Comment:
`target_transform` should be applied to the dataset before splitting

Reply:
ok, that makes sense.